Automattic/wp-calypso
# Replace isWooCoreProfilerFlow with isPasswordlessJetpackConnectionFlow

Related to https://github.com/Automattic/wp-calypso/pull/95041

This is an enhanced version of https://github.com/Automattic/wp-calypso/pull/95041 

## Proposed Changes

* Replaces `isWooCoreProfilerFlow` with `isPasswordlessJetpackConnectionFlow`. This is a more abstract approach 
to determining whether to display the new Jetpack connection page. `isPasswordlessJetpackConnectionFlow` [checks](https://github.com/Automattic/wp-calypso/compare/update/refactor-to-add-wcpay-to-passwordless-core-profiler?expand=1#diff-3e6a74b418f2e50061e081a1c3209fe3f6df30522176d876427e8f8e6568c675R25) for `isWooCoreProfilerFlow` and `isWooCommercePaymentsOnboardingFlow`


## Testing Instructions

1. Create a new JN site with this latest woocommerce.
2. Go through the core profiler and select `Jetpack` on the extensions page.
3. Continue and you should be redirected to the Jetpack connection page.
4. Change `from` to `woocommerce-payments` and add `plugin_name=woocommerce-payments`  param to the URL and then copy the URL.
5. Checkout this branch locally and set it up with [woocommerce-start-dev-env](https://github.com/woocommerce/woocommerce-start-dev-env)
6. Visit the URL
7. You should still see the new Jetpack connection page.

## Pre-merge Checklist
